paper-api/src/main/java/io/papermc/paper/configuration/ServerConfiguration.java

```diff
@@ -0,0 +1,46 @@
+package io.papermc.paper.configuration;
+
+/**
+ * Represents the configuration settings for a server.
+ * <p>
+ * This interface doesn't aim to cover every possible server configuration
+ * option but focuses on selected critical settings.
+ */
+public interface ServerConfiguration {
+
+    /**
+     * Gets whether the server is in {@link org.bukkit.Server#getOnlineMode() online mode}
+     * or behind a proxy that uses online mode.
+     *
+     * @return true if the server is in proxied online mode, false otherwise
+     */
+    boolean isProxyOnlineMode();
+
+    /**
+     * Gets whether the server is configured for a Velocity proxy.
+     *
+     * @return true if the server has Velocity enabled, false otherwise
+     */
+    boolean isVelocityEnabled();
+
+    /**
+     * Gets whether the server is configured for a BungeeCord proxy.
+     *
+     * @return true if the server has BungeeCord enabled, false otherwise
+     */
+    boolean isBungeeCordEnabled();
+
+    /**
+     * Gets whether the server is configured for Velocity online mode.
+     *
+     * @return true if the server is in Velocity online mode, false otherwise
+     */
+    boolean isVelocityOnlineMode();
+
+    /**
+     * Gets whether the server is configured for BungeeCord online mode.
+     *
+     * @return true if the server is in BungeeCord online mode, false otherwise
+     */
+    boolean isBungeeCordOnlineMode();
```

Comment:
I feel like it would be worth to add a note that these also check if the proxy mode is enabled to begin with, given both return only true if their proxy mode is enabled and in online mode.